app/src/main/java/com/hedvig/app/feature/marketpicker/MarketAdapter.kt

```diff
@@ -0,0 +1,115 @@
+package com.hedvig.app.feature.marketpicker
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.RadioButton
+import android.widget.TextView
+import androidx.dynamicanimation.animation.SpringAnimation
+import androidx.dynamicanimation.animation.SpringForce
+import androidx.recyclerview.widget.RecyclerView
+import com.hedvig.app.R
+import com.hedvig.app.util.extensions.compatDrawable
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import com.hedvig.app.util.spring
+import kotlinx.android.synthetic.main.market_item.view.*
+
+class MarketAdapter(private val model: MarketPickerViewModel, private val marketId: Int) :
+    RecyclerView.Adapter<MarketAdapter.ViewHolder>() {
+
+    private var lastChecked: RadioButton? = null
+    private var lastCheckedPos = 0
```

Comment:
I think we can let the ViewModel own the selection state in addition to owning the available markets here: See `CategoriesAdapter` for inspiration 🙂